engine/src/main/java/org/pentaho/di/job/entries/setvariables/JobEntrySetVariables.java

```diff
@@ -255,9 +255,12 @@ public Result execute( Result result, int nr ) throws KettleException {
 
       if ( variableName != null ) {
         for ( int i = 0; i < variableName.length; i++ ) {
-          variables.add( variableName[ i ] );
-          variableValues.add( variableValue[ i ] );
-          variableTypes.add( variableType[ i ] );
+          if ( ( variables.contains( variableName[ i ] ) && replaceVars ) || !variables.contains(
```

Comment:
Is there a Testrail or Practitest to capture this case? 
Or is there been an existing Testrail or Practitest that needs to be updated?

Reply:
Added a new test case to cover this testJobEntrySetVariablesExecute_VARIABLE_TYPE_ROOT_JOB_VariableNotNull in JobEntrySetVariablesTest class